app/services/dttp/defer.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Dttp
+  class Defer
+    include ServicePattern
+    include Mappable
+
+    class Error < StandardError; end
+
+    attr_reader :trainee
+
+    def initialize(trainee:)
+      @trainee = trainee
+    end
+
+    def call
+      body = { "dfe_TraineeStatusId@odata.bind" => "/dfe_traineestatuses(#{dttp_status_id('Trainee deferred')})" }
+
+      dttp_update("/dfe_placementassignments(#{trainee.placement_assignment_dttp_id})", body)
+
+      trainee.defer!
```

Comment:
I think we should do these state changes when we have queued the job rather than when the job has succeeded. If the job fails or if there is a bit of a lag with the queue the UI stays the same so the user will likely think it didn't work and do the action (defer, submit etc) again and we'll get multiple jobs queued for the same task.(particularly bad when create a new trainee but we should avoid it in general). If the job fails the user has no recourse to fix it anyway and it will be up to us to investigate the error so I think we should move the state transitions further forward in the process.

Reply:
Makes sense! Especially since "if the job fails the user has no recourse to fix it".

There are already other state changes that happen within the jobs themselves, so it probably makes sense to tackle these all together. @gpeng - is there a card for this in the next sprint or should I add one?